addons/mail/sfu_server/src/models/channel.js

```diff
@@ -0,0 +1,224 @@
+            throw new Error(`channel [${uuid}] does not exist`);
+        }
+        const session = channel.join(sessionId);
+        return { channel, session };
+    }
+
+    static closeAll() {
+        for (const channel of Channel.records.values()) {
+            channel.close();
+        }
+        Channel.records.clear();
+    }
+
+    /**
+     * @returns {{ channelCount: number, sessionCount: number}}
+     */
+    static getStats() {
+        return {
+            channelCount: Channel.records.size,
+            sessionCount: [...Channel.records.values()].reduce(
+                (acc, channel) => acc + channel.sessions.size,
+                0
+            ),
+        };
+    }
+
+    /**
+     * @param {string} remoteAddress
+     * @param {Object} param1
+     * @param {import("mediasoup").types.Worker} param1.worker
+     * @param {import("mediasoup").types.Router} param1.router
+     */
+    constructor(remoteAddress, { worker, router }) {
+        super();
+        this.remoteAddress = remoteAddress;
+        this.uuid = crypto.randomUUID();
+        this.name = `${remoteAddress}*${this.uuid.slice(-5)}`;
+        this.router = router;
+        this._worker = worker;
+        this._onSessionClose = this._onSessionClose.bind(this);
+        Channel.records.set(this.uuid, this);
+    }
+
+    /**
+     * @return {Promise<{total: number, screen: number, audio: number, camera: number}>}
+     */
+    async getIncomingBitrate() {
+        let audioSum = 0;
+        let cameraSum = 0;
+        let screenSum = 0;
+        for (const session of this.sessions.values()) {
+            const { audio, camera, screen } = await session.getProducerBitRates();
+            audioSum += audio || 0;
+            cameraSum += camera || 0;
+            screenSum += screen || 0;
+        }
+        return {
+            audio: audioSum,
+            camera: cameraSum,
+            screen: screenSum,
+            total: audioSum + cameraSum + screenSum,
+        };
+    }
+
+    /**
+     * @returns {Object<number, Object>}
+     */
+    get sessionsInfo() {
+        const sessionsInfo = {};
+        for (const session of this.sessions.values()) {
+            sessionsInfo[session.id] = session.info;
+        }
+        return sessionsInfo;
+    }
+
+    /**
+     * @returns {import("mediasoup").types.WebRtcServer}
+     */
+    get webRtcServer() {
+        return this._worker.appData.webRtcServer;
+    }
+
+    /**
+     * @param {number} sessionId
+     * @fires Channel#sessionJoin
+     * @returns {Session}
+     */
+    join(sessionId) {
+        const oldSession = this.sessions.get(sessionId);
+        if (oldSession) {
+            oldSession.off("close", this._onSessionClose);
+            oldSession.close(SESSION_CLOSE_CODE.REPLACED);
+        }
+        const session = new Session(sessionId, this);
+        this.sessions.set(session.id, session);
+        session.once("close", this._onSessionClose);
+        this.setCloseTimeout(false);
```

Comment:
This bypasses the one person limit if the first person who joins just stays alone forever